packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
+
+export interface CoreDashboardModel {
+  panels: CorePanelModel[];
+  uid: string;
+  title: string;
+  updatePanels: (panels: CorePanelModel[]) => unknown;
+}
+
+export interface CoreDashboardSrv {
+  dashboard?: CoreDashboardModel;
+}
```

Comment:
not sure I understand why we need this interface

Reply:
Because we are creating wrappers of the core objects we can't have the wrappers believe they are the core objects itself or we'll have to start dealing with many type problems.

For example. the `DashboardSrv` defined in grafana-core has a `dashboard` attribute but it doesn't have a `getCurrentDashboard` method. so if grafana calls `setDashboardSrv` inside the core App.ts and this `setDashboardSrv` expects to accept the same interface that the user will get, typescript will throw errors because of that missing method.

So this interface is a middle-step between the internal (core) objects and the public ones.

Reply:
> When the scenes architecture is in place we should be able to come up with a much better and powerful api that panels / plugins can use to interact with the current dashboard.

IMO we should better do that when the architecture is in place and ready to replace the old one. We have already in our APIs a lot of deprecated things that have no replacements. Marking things deprecated without a clear replacement is very confusing as a developer.

We can add remarks to the jsdoc indicating that this API will be replaced soon, but I rather avoid the deprecated noise if we can.

Reply:
Resolution: We are not publishing this new API as deprecated until a new ready to use API is in place to replace it.